gameofgame.py

```diff
@@ -25,49 +25,65 @@ def choose_country():
 # Function to check if the guess is correct
 def check_guess(guess, country):
     capital = countries_capitals[country]
-
     if guess.lower() == capital.lower():
-    return True
+        return True
     else:
-    return False
- main
+        return False
+
+# Initialize player scores
+player1_score = 0
+player2_score = 0
 
 # Main game loop
-    while True:
+while True:
     # Choose a random country
     country = choose_country()
 
     # Both Players take turns to play
-    #player1 goes first
+    # player 1 goes first
     print("Player 1's turn")
     player1_guess = input(f"What is the capital of {country}? ")
     
-    ##player2
-    print ("Player 2's turn")
+    # player 2 goes next
+    print("Player 2's turn")
     player2_guess = input(f"What is the capital of {country}? ")
 
-    ##Reveal the correct answer
+    # Reveal the correct answer
     capital = countries_capitals[country]
-    print (f"The correct answer is {capital}.")
-    
+    print(f"The correct answer is {capital}.")
 
-     ##check the guesses and print results
+    # Check the guesses and update the scores
     if check_guess(player1_guess, country) and check_guess(player2_guess, country):
         print("Both players guessed correctly!")
+        player1_score += 1
+        player2_score += 1
     elif check_guess(player1_guess, country):
         print("Player 1 guessed correctly!")
         print("Well done, you know your Geography well!")
+        player1_score += 1
     elif check_guess(player2_guess, country):
         print("Player 2 guessed correctly!")
         print("Well done, you know your Geography well!")
+        player2_score += 1
     else:
         print("Neither player guessed correctly.")
-        print("You both should learn about capital cities in the world!")
-        
+        print("You both should learn about capital cities in the world!!")
         
+    # Print the current scores
+    print(f"Current scores: Player 1: {player1_score}, Player 2: {player2_score}")
+    
     # Ask if the players want to continue playing
     play_again = input("Do you want to play again? (y/n) ")
     if play_again.lower() != "y":
         break
     
+# Print the final scores and congratulate the winner
+print(f"Final scores: Player 1: {player1_score}, Player 2: {player2_score}")
+if player1_score > player2_score:
+    print("Congratulations Player 1, you are the winner!")
+elif player2_score > player1_score:
+    print("Congratulations Player 2, you are the winner!")
+else:
+    print("The game ended in a tie. Well played both players!")
```

Comment:
Here's a suggestion for how to refactor this section to avoid duplication of messages:

```suggestion
congrats_message = "Congratulations Player {winner}, you are the winner!"
if player1_score > player2_score:
    print(congrats_message.format(winner=1)
elif player2_score > player1_score:
    print(congrats_message.format(winner=2)
else:
    print("The game ended in a tie. Well played both players!")
```